Starting from version 1.20.2 the ping applet doesn't work properly, only the first sent packet
get a response. Seems that starting from the second icmp, the packet checksum is wrong.
This patch revert the ping.c code as in version 1.19.4, despite the fact that this isn't the
right solution it simply works.

--- busybox-1.21.0/networking/ping.c.vanilla	2013-06-27 17:20:36.781685499 +0000
+++ busybox-1.21.0/networking/ping.c	2013-06-27 17:21:31.296112213 +0000
@@ -149,6 +149,31 @@
 	PINGINTERVAL = 1, /* 1 second */
 };
 
+/* Common routines */
+
+static int in_cksum(unsigned short *buf, int sz)
+{
+	int nleft = sz;
+	int sum = 0;
+	unsigned short *w = buf;
+	unsigned short ans = 0;
+
+	while (nleft > 1) {
+		sum += *w++;
+		nleft -= 2;
+	}
+
+	if (nleft == 1) {
+		*(unsigned char *) (&ans) = *(unsigned char *) w;
+		sum += ans;
+	}
+
+	sum = (sum >> 16) + (sum & 0xFFFF);
+	sum += (sum >> 16);
+	ans = ~sum;
+	return ans;
+}
+
 #if !ENABLE_FEATURE_FANCY_PING
 
 /* Simple version */
@@ -176,7 +201,7 @@
 	pkt = (struct icmp *) G.packet;
 	memset(pkt, 0, sizeof(G.packet));
 	pkt->icmp_type = ICMP_ECHO;
-	pkt->icmp_cksum = inet_cksum((uint16_t *) pkt, sizeof(G.packet));
+	pkt->icmp_cksum = in_cksum((unsigned short *) pkt, sizeof(G.packet));
 
 	xsendto(pingsock, G.packet, DEFDATALEN + ICMP_MINLEN, &lsa->u.sa, lsa->len);
 
@@ -468,7 +493,7 @@
 		/* No hton: we'll read it back on the same machine */
 		*(uint32_t*)&pkt->icmp_dun = monotonic_us();
 
-	pkt->icmp_cksum = inet_cksum((uint16_t *) pkt, datalen + ICMP_MINLEN);
+	pkt->icmp_cksum = in_cksum((unsigned short *) pkt, datalen + ICMP_MINLEN);
 
 	sendping_tail(sendping4, ICMP_MINLEN);
 }
@@ -487,7 +512,7 @@
 	/*if (datalen >= 4)*/
 		*(uint32_t*)(&pkt->icmp6_data8[4]) = monotonic_us();
 
-	//TODO? pkt->icmp_cksum = inet_cksum(...);
+	//TODO? pkt->icmp_cksum = in_cksum(...);
 
 	sendping_tail(sendping6, sizeof(struct icmp6_hdr));
 }
